integration-cli/docker_cli_events_test.go

```diff
@@ -203,68 +204,42 @@ func TestEventsImageImport(t *testing.T) {
+		events = events[:len(events)-1]
+		for _, event := range events {
+			eventFields := strings.Fields(event)
+			eventName := eventFields[len(eventFields)-1]
+			if ok, err := regexp.MatchString(match, eventName); err != nil || !ok {
+				t.Fatalf("event should match %s, got %#v, err: %v", match, eventFields, err)
+			}
+		}
 	}
-	defer d.Stop()
 
 	since := time.Now().Unix()
-	out, err := d.Cmd("run", "--rm", "busybox", "true")
+	out, _, err := runCommandWithOutput(exec.Command(dockerBinary, "run", "--rm", "busybox", "true"))
 	if err != nil {
 		t.Fatal(out, err)
 	}
-	out, err = d.Cmd("run", "--rm", "busybox", "true")
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "run", "--rm", "busybox", "true"))
 	if err != nil {
 		t.Fatal(out, err)
 	}
-	out, err = d.Cmd("events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die")
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die"))
 	if err != nil {
 		t.Fatalf("Failed to get events: %s", err)
 	}
-	events := strings.Split(out, "\n")
-	events = events[:len(events)-1]
-	if len(events) != 2 {
-		t.Fatalf("Expected 2 events, got %d: %v", len(events), events)
-	}
-	dieEvent := strings.Fields(events[len(events)-1])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
-	}
+	parseEvents(out, "die")
 
-	dieEvent = strings.Fields(events[len(events)-2])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
-	}
-
-	out, err = d.Cmd("events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die", "--filter", "event=start")
+	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "events", fmt.Sprintf("--since=%d", since), fmt.Sprintf("--until=%d", time.Now().Unix()), "--filter", "event=die", "--filter", "event=start"))
 	if err != nil {
 		t.Fatalf("Failed to get events: %s", err)
 	}
-	events = strings.Split(out, "\n")
-	events = events[:len(events)-1]
-	if len(events) != 4 {
-		t.Fatalf("Expected 4 events, got %d: %v", len(events), events)
-	}
-	startEvent := strings.Fields(events[len(events)-4])
-	if startEvent[len(startEvent)-1] != "start" {
-		t.Fatalf("event should be start, not %#v", startEvent)
-	}
-	dieEvent = strings.Fields(events[len(events)-3])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
-	}
-	startEvent = strings.Fields(events[len(events)-2])
-	if startEvent[len(startEvent)-1] != "start" {
-		t.Fatalf("event should be start, not %#v", startEvent)
-	}
-	dieEvent = strings.Fields(events[len(events)-1])
-	if dieEvent[len(dieEvent)-1] != "die" {
-		t.Fatalf("event should be die, not %#v", dieEvent)
+	parseEvents(out, "((die)|(start))")
+
+	// make sure we at least got 2 start events
+	if strings.Count(out, "start") < 2 {
```

Comment:
because we started two? I'm kinda nervous about setting an exact value
because thats the rabbit hole we went down before

On Tue, Feb 24, 2015 at 6:14 PM, Alexander Morozov <notifications@github.com
> wrote:

> In integration-cli/docker_cli_events_test.go
> <https://github.com/docker/docker/pull/10998#discussion_r25313215>:
>
> > -	}
> > -	dieEvent = strings.Fields(events[len(events)-3])
> > -	if dieEvent[len(dieEvent)-1] != "die" {
> > -		t.Fatalf("event should be die, not %#v", dieEvent)
> > -	}
> > -	startEvent = strings.Fields(events[len(events)-2])
> > -	if startEvent[len(startEvent)-1] != "start" {
> > -		t.Fatalf("event should be start, not %#v", startEvent)
> > -	}
> > -	dieEvent = strings.Fields(events[len(events)-1])
> > -	if dieEvent[len(dieEvent)-1] != "die" {
> > -		t.Fatalf("event should be die, not %#v", dieEvent)
> > +	parseEvents(out, "((die)|(start))")
> > +
> > +	// make sure we at least got 2 start events
> > +	if strings.Count(out, "start") < 2 {
>
> Why at least? :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/10998/files#r25313215>.
>